### PR TITLE
[WIP] - [Fix Bug 1277196] Decommission /firstrun/learnmore/ pages and /window…

### PR DESCRIPTION
**Pull request by RCMainak**
## Description
Decommission /firstrun/learnmore/ and /windows-10/welcome/ pages. Removing all the unnecessary assets used by those files. Adding redirects to appropriate pages.  
## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1277196
## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing.

### CODE REVIEW

**Comment by RCMainak** [comment]
This is what i've done so far. Any feedback on where I'm heading in the wrong direction!!! :wink:

**Comment by alexgibson** [comment]
Putting a do-not-merge on this as it needs to wait until after Firefox 50 is released.

File: tests/redirects/map_globalconf.py [hunk 1]
```diff
@@ -1094,4 +1094,9 @@
 
     # Bug 1313023
     url_test('/story', 'https://donate.mozilla.org/?source=story_redirect'),
+
+    # Bug 1277196
+    url_test('/firefox/firstrun/learnmore', 'firefox/features'),
```
Comment (alexgibson) [inline tests/redirects/map_globalconf.py]: This should be `/firefox/features/`.

File: bedrock/firefox/redirects.py [hunk 2]
```diff
@@ -520,4 +520,8 @@ def firefox_mobile_faq(request, *args, **kwargs):
 
     # bug 1299947
     redirect('^firefox/channel/?$', 'firefox.channel.desktop'),
+
+    # Bug 1277196
+    redirect('^firefox/firstrun/learnmore/$', 'firefox.features'),
+    redirect('^firefox/windows-10/welcome/$', 'https://support.mozilla.org/kb/how-change-your-default-browser-windows-10'),
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py]: You need to make sure the last trailing slash is optional here: `r'^firefox/windows-10/welcome/?$'`

File: bedrock/firefox/redirects.py [hunk 3]
```diff
@@ -520,4 +520,8 @@ def firefox_mobile_faq(request, *args, **kwargs):
 
     # bug 1299947
     redirect('^firefox/channel/?$', 'firefox.channel.desktop'),
+
+    # Bug 1277196
+    redirect('^firefox/firstrun/learnmore/$', 'firefox.features'),
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py]: This also needs an optional trailing slash: `r'^firefox/firstrun/learnmore/?$'`

Comment (alexgibson) [inline bedrock/firefox/redirects.py]: In addition to this redirect, we also need to handle the same URL with a version number e.g. https://www.mozilla.org/en-US/firefox/50.0/firstrun/learnmore/

It's probably easiest to add another redirect for this, using a regexp for the version number:

```
redirect(r'^firefox(?:\/\d+\.\d+(?:\.\d+)?(?:a\d+)?)?/firstrun/learnmore/?$', 'firefox.features'),
```

A test to accompany this extra test would be good too.

```
url_test('/firefox/{49.0,49.0.1,50.0a1,51.0a2}/firstrun/learnmore', '/firefox/features/'),
```

**Comment by alexgibson** [comment]
Please delete these two functional test files too:

1. https://github.com/mozilla/bedrock/blob/master/tests/functional/firefox/test_win10_welcome.py
2. https://github.com/mozilla/bedrock/blob/master/tests/pages/firefox/win10_welcome.py

**Comment by alexgibson** [comment]
You can also delete these JS files:

1. https://github.com/mozilla/bedrock/blob/master/media/js/firefox/win10-welcome-init.js
2. https://github.com/mozilla/bedrock/blob/master/media/js/firefox/win10-welcome.js
3. https://github.com/mozilla/bedrock/blob/master/media/js/firefox/firstrun/learnmore/yahoo-search.js
4. https://github.com/mozilla/bedrock/blob/master/media/js/firefox/firstrun/learnmore/learnmore.js
5. https://github.com/mozilla/bedrock/blob/master/tests/unit/spec/firefox/win10-welcome.js

**Comment by alexgibson** [comment]
You can delete the following bundles from `bedrock/settings/static_media.py`:

1. `firefox_firstrun_42_learnmore` JS bundle
2. `firefox_firstrun_learnmore_yahoo_search` CSS & JS bundles
3. `firefox_win10_welcome` CSS and JS bundles
4. `firefox_win10_welcome_variants` CSS bundle.

**Comment by RCMainak** [comment]
with pleasure :stuck_out_tongue_winking_eye: . Nice to see that I've added the required things properly :smile:

**Comment by RCMainak** [comment]
@alexgibson Do I need to remove `firefox_firstrun_ravioli` which can be found [here](https://github.com/mozilla/bedrock/blob/master/bedrock/settings/static_media.py#L382-L387)??

**Comment by alexgibson** [comment]
> Do I need to remove firefox_firstrun_ravioli which can be found here?? 

Nope that's used in a [different template](https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/firstrun/ravioli.html) unrelated to this bug. Thanks

**Comment by RCMainak** [comment]
well, I've deleted some ravioli related resources previously. Can you check on those once again? Please, just to make sure everything fine so far :smile:  @alexgibson

**Comment by alexgibson** [comment]
> well, I've deleted some ravioli related resources previously. Can you check on those once again? Please, just to make sure everything is fine so far 😄 @alexgibson

Just because they are named ravioli doesn't mean they are related to this change. I'll give things another check once you make the amends I've pointed out so far.

File: tests/redirects/map_globalconf.py [hunk 4]
```diff
@@ -1094,4 +1094,9 @@
 
     # Bug 1313023
     url_test('/story', 'https://donate.mozilla.org/?source=story_redirect'),
+
+    # Bug 1277196
+    url_test('/firefox/{49.0,49.0.1,50.0a1,51.0a2}/firstrun/learnmore', 'firefox/features/'),
```
Comment (alexgibson) [inline tests/redirects/map_globalconf.py]: I think this needs to be `/firefox/features/`

Comment (alexgibson) [inline tests/redirects/map_globalconf.py]: Please also make sure we have a test for the URL without a version number (just `/firefox/firstrun/learnmore`). You seem to have deleted the original redirect for this.

Comment (RCMainak) [inline tests/redirects/map_globalconf.py]: Oh my dear god :fearful:

**Comment by alexgibson** [comment]
I found a [page here](https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/firstrun/ravioli.html#L46) that's using the `/firstrun/learnmore/yahoo/firefox-logo.png` asset. We should make it so this page is not relying on anything under `/learnmore/`. We could probably change the page to use a logo from the [shared firefox asset directory](https://github.com/mozilla/bedrock/tree/master/media/img/firefox/template) instead.

**Comment by alexgibson** [comment]
Please reinstate both of these assets, which are not related to this change:

- `media/img/firefox/firstrun/ravioli.png`
- `media/img/firefox/firstrun/ravioli-high-res.png`

**Comment by alexgibson** [comment]
Please remove the two assets from the Karma test runner [configuration file](https://github.com/mozilla/bedrock/blob/master/tests/unit/karma.conf.js):

- `'media/js/firefox/win10-welcome.js'`
- `'tests/unit/spec/firefox/win10-welcome.js'`

**Comment by alexgibson** [comment]
Looking really good, just a few more loose ends to tidy up 👍

File: bedrock/firefox/templates/firefox/firstrun/ravioli.html [hunk 5]
```diff
@@ -43,7 +43,7 @@
 <main role="main" class="container">
   <h2>
     <a href="{{ url('firefox.new') }}">
-      {{ high_res_img('firefox/firstrun/learnmore/yahoo/firefox-logo.png', {'alt': 'Firefox', 'width': '97', 'height': '100'}) }}
+      {{ high_res_img('firefox/template/logo.png', {'alt': 'Firefox', 'width': '97', 'height': '100'}) }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/firstrun/ravioli.html]: `logo-large.png` would be a slightly better fit here.

Comment (RCMainak) [inline bedrock/firefox/templates/firefox/firstrun/ravioli.html]: roger that :guardsman:

**Comment by alexgibson** [comment]
I confirmed via a new profile that Firefox 50.0 no longer opens `/firstrun/learnmore/`

**Comment by alexgibson** [comment]
r+, thanks for taking this on @RCMainak 💯

**Pull request merged**